src/microbe_stage/systems/MucocystSystem.cs

```diff
@@ -0,0 +1,60 @@
+﻿namespace Systems;
+
+using System;
+using Components;
+using DefaultEcs;
+using DefaultEcs.System;
+using DefaultEcs.Threading;
+
+/// <summary>
+///   Handles gaining immortality from mucocyst ability
+/// </summary>
+[With(typeof(MicrobeControl))]
+[With(typeof(Health))]
+[With(typeof(CompoundStorage))]
+[Without(typeof(AttachedToEntity))]
+[WritesToComponent(typeof(CellProperties))]
+[RunsAfter(typeof(OrganelleComponentFetchSystem))]
+[RunsBefore(typeof(MicrobeMovementSystem))]
+public sealed class MucocystSystem : AEntitySetSystem<float>
+{
+    private Compound mucilageCompound = SimulationParameters.Instance.GetCompound("mucilage");
+
+    public MucocystSystem(World world, IParallelRunner runner) :
+        base(world, runner)
+    {
+    }
+
+    protected override void Update(float delta, in Entity entity)
+    {
+        // Handles invulnerability from mucocyst. Other buffs/debuffs from mucocyst are in related systems
+        ref var control = ref entity.Get<MicrobeControl>();
+
+        if (control.State == MicrobeState.MucocystShield)
+        {
+            ref var storage = ref entity.Get<CompoundStorage>();
+
+            storage.Compounds.
+                Compounds[mucilageCompound] -= Math.Min(
+                Constants.MUCOCYST_MUCILAGE_DRAIN * delta,
+                storage.Compounds.Compounds[mucilageCompound]);
+
+            if (storage.Compounds.Compounds[mucilageCompound] <= 0)
+                control.State = MicrobeState.Normal;
+
+            entity.Get<Health>().Invulnerable = true;
+
+            var membrane = entity.Get<CellProperties>().CreatedMembrane;
```

Comment:
As this writes to membrane this must have the runs on main thread attribute. And also as this just affects the mucocyst parameters it would be enough to mark this as running on entities with CellProperties but just in read mode:

```
[With(typeof(CellProperties))]
[ReadsComponent(typeof(CellProperties)]
```